sdk/resourcemanager/internal/testutil/recording.go

```diff
@@ -107,37 +124,71 @@ func StartRecording(t *testing.T, pathToPackage string) func() {
 
 // StopRecording stops the recording.
 func StopRecording(t *testing.T) {
-	err := recording.Stop(t, &recording.RecordingOptions{Variables: map[string]interface{}{recordingRandomSeedVariableName: strconv.FormatInt(recordingRandomSeed, 10)}})
+	err := recording.Stop(t, &recording.RecordingOptions{Variables: map[string]interface{}{recordingRandomSeedVariableName: strconv.FormatInt(recordingSeed, 10)}})
 	if err != nil {
 		t.Fatalf("Failed to stop recording: %v", err)
 	}
 }
 
+func initRandomSource(t *testing.T) {
+
+	if recordingRandomSource != nil {
+		return
+	}
+
+	var seed int64
+	var err error
+
+	variables := recording.GetVariables(t)
+	seedString, ok := variables[recordingRandomSeedVariableName]
+	if ok {
+		seed, err = strconv.ParseInt(seedString.(string), 10, 64)
+	}
+
+	// We did not have a random seed already stored; create a new one
+	if !ok || err != nil || recording.GetRecordMode() == "live" {
+		seed = time.Now().Unix()
+		val := strconv.FormatInt(seed, 10)
+		variables[recordingRandomSeedVariableName] = &val
+	}
+
+	// create a Source with the seed
+	recordingRandomSource = rand.NewSource(seed)
+	recordingSeed = seed
+}
+
 // GenerateAlphaNumericID will generate a random alpha numeric ID.
 // When handling live request, the random seed is generated.
 // Otherwise, the random seed is stable and will be stored in recording file.
 // The length parameter is the random part length, not include the prefix part.
 func GenerateAlphaNumericID(t *testing.T, prefix string, length int) string {
-	if recording.GetRecordMode() != "live" {
-		if recordingRandomSeed == 0 {
-			variables := recording.GetVariables(t)
-			if seed, ok := variables[recordingRandomSeedVariableName]; ok {
-				seedNum, err := strconv.ParseInt(seed.(string), 10, 64)
-				if err == nil {
-					recordingRandomSeed = seedNum
-				}
+
+	var lowercaseOnly bool = false
+
+	initRandomSource(t)
+	sb := strings.Builder{}
+	sb.Grow(length)
+	sb.WriteString(prefix)
+	i := length - len(prefix) - 1
```

Comment:
Described in the function doc, the length does not include the prefix part.

Reply:
I can switch this now so that nothing breaks.  However when you update the mgmt plane to use the recording framework in sdk/internal you will need to update all recordings.  How the code is implemented is how the shared recording infrastructure works.